src/App.jsx

```diff
@@ -0,0 +1,257 @@
+import { useEffect, useState } from 'react';
+
+import './App.css';
+import { constants } from './app/config';
+import { getDirection, getMoveStepPositions } from './app/game';
+import { useStateValue } from './app/state/useStateValue';
+import Dice from './components/dice/Dice';
+import Grid from './components/grid/Grid';
+import { arrayIsInArray, log, randomNumber } from './utils/main';
+
+const App = () => {
+  // hooks
+  const [state, dispatch] = useStateValue();
+  const [uiDisabled, setUIDisabled] = useState(false);
+  const [isBootleTurn, setIsBottleTurn] = useState(false);
+
+  // effects
+  useEffect(() => {
+    if (state.gameStatus.running && isBootleTurn) {
+      onClickRollDiceHandler();
```

Comment:
I debug the code

and all intervals are cleared after players moving, none is stalled...and don't have any side effect
(I leave a variable const debugTimeInterval = false, that if enabled we can observe it)
...this is managed inside onClickRollDiceHandler setInterval when it detects player stop's

I understand that this must be a hack way and not a hooks idiomatic way, 
but to implement this delay in player movements, like a sort of animation, I must use a setInterval and in this case conflicts with your wise recommendation and best practices (not leave stalled intervals running)

but if I clear the interval in useEffect cleanup the player stop because, we clear the interval, 
seems that the solution is not use timers to move players without delay, and use a simple loop without any sort of delay, or thrust that clear intervals are well implemented in logic

I completely agree that this is far from optional..........works but not perfect

or try to figure out a approach that works in both cases :)

do you have any suggestion @buraktt 

what I think is a requirement, get rid of timer and move players without delay, but in my humble opinion I like the way delay is used, else players will very fast

waiting for your advice and recommendation in this recommendation

Reply:
Hello Burak

I already do the clean up in the weekend, like you suggested, but if I clean up the interval on unmount, player movement stops because it is controlled by that time interval

I try to explain that in push review comment, if I clear interval, stop occurs, and game stop creating other side effects,  that's why I request some advice

For what I see in that debug, the only way to get rid of set interval is refactor all code and move the players from position x to y, with one step only, and lost this delays and animation, like player moves one step from x to x milliseconds......

I don't know how others that do the challenge did the delay and animation without unmount, and without timmers...to be honest I can't figure it out
Maybe use the timer outside of component in game.js

I can try refactor and get rid of interval in component, and start from that point to see if some idea occurs

About memory leaks, I debug everything and when players stop moving, all intervals are clear, not in unmounts but when players stop move

I understand that this is a hack and to far from ideal

Waiting from some advice here

Like refactor code, or get rid of interval in component

Thanks

Reply:
FIXED with great recommendation of @realiarthur